ir/src/values/mod.rs

```diff
@@ -41,6 +41,19 @@ pub enum Integer {
     I128(i128),
 }
 
+impl Integer {
+    pub fn get_unsigned(&self) -> u32 {
+        match self {
+            Self::U8(n) => *n as u32,
+            Self::U16(n) => *n as u32,
+            Self::U32(n) => *n,
+            Self::U64(n) => u32::try_from(*n).unwrap_or_else(|e| panic!("{}: {:?}", e, self)),
+            Self::U128(n) => u32::try_from(*n).unwrap_or_else(|e| panic!("{}: {:?}", e, self)),
```

Comment:
Just to confirm (without context), is downcasting the intended behavior here?

Reply:
to offer context: i wrote it so that array ranges could properly calculate intermediate slice length during complex array access (ie `x[0..6][0..2][1]`). it panics if the int is bigger than u32::MAX because array ranges use u32 for len.

that being said, yes it is essentially just downcasting to a u32. ill move it to a downcast impl now. thanks!